apps/mobile/src/features/swap/swap-state/utils/secondary-amount.ts

```diff
@@ -0,0 +1,41 @@
+import { SecondaryAmount, SwapInternalState } from '@/features/swap/swap-state/swap-state.types';
+import { whenInputCurrencyMode } from '@/utils/when-currency-input-mode';
+
+import { Money } from '@leather.io/models';
+
+interface ComputeSecondaryAmountStateParams {
+  state: SwapInternalState;
+  queryStatus: 'pending' | 'success' | 'error';
+  isFetching: boolean;
+  derivedAmounts: { crypto: Money | null; quote: Money | null };
```

Comment:
The `derivedAmounts` type is repeated a few times, maybe it can be refactored to be shared?